nemo_skills/pipeline/train.py

```diff
@@ -31,6 +32,135 @@
+    extra_arguments = (
+        f" ++model.data.train_ds.file_path='{params.training_data}' "
+        f" ++model.data.validation_ds.file_path='{params.validation_data}' "
+        f" model.restore_from_path={params.nemo_model} " + params.extra_arguments
+    )
+    cmd = (
+        f"export WANDB_API_KEY={os.getenv('WANDB_API_KEY', '')} && "
+        f"export HF_TOKEN={get_token()} && "
+        f"export HYDRA_FULL_ERROR=1 && "
+        f"export PYTHONPATH=$PYTHONPATH:/nemo_run/code && "
+        f"cd /nemo_run/code && "
+        f"echo 'Starting training' && "
+        f"{training_script} "
+        f"    {config_params}"
+        f"    ++model.tensor_model_parallel_size={params.num_gpus} "
+        f"    trainer.devices={params.num_gpus} "
+        f"    trainer.num_nodes={params.num_nodes} "
+        f"    {params.logging_params} "
+        f"    exp_manager.name={params.expname} "
+        f"    exp_manager.explicit_log_dir={params.output_dir}/training "
+        f"    exp_manager.exp_dir={params.output_dir}/training "
+        f"    ++exp_manager.max_time_per_run={params.timeout} "
+        f"    {extra_arguments} "
+    )
+    return cmd
+
+
+def train_dpo(params: TrainingParams) -> str:
+    if params.config_name is None:
+        config_name = "dpo_config"
+    else:
+        config_name = params.config
+    config_params = f"--config-name={config_name} --config-path={params.config_path} "
+    training_script = "python -m nemo_skills.training.start_dpo"
+    extra_arguments = (
+        f" ++model.data.data_prefix.train='[{params.training_data}]' "
+        f" ++model.data.data_prefix.validation='[{params.validation_data}]' "
+        f" ++model.data.data_prefix.test='[{params.validation_data}]' "
+        f" pretrained_checkpoint.restore_from_path={params.nemo_model} " + params.extra_arguments
+    )
+    cmd = (
+        f"export WANDB_API_KEY={os.getenv('WANDB_API_KEY', '')} && "
+        f"export HF_TOKEN={get_token()} && "
+        f"export HYDRA_FULL_ERROR=1 && "
+        f"export PYTHONPATH=$PYTHONPATH:/nemo_run/code && "
+        f"cd /nemo_run/code && "
+        f"echo 'Starting training' && "
+        f"{training_script} "
+        f"    {config_params}"
+        f"    ++model.tensor_model_parallel_size={params.num_gpus} "
+        f"    trainer.devices={params.num_gpus} "
+        f"    trainer.num_nodes={params.num_nodes} "
+        f"    {params.logging_params} "
+        f"    exp_manager.name={params.expname} "
+        f"    exp_manager.explicit_log_dir={params.output_dir}/training "
+        f"    exp_manager.exp_dir={params.output_dir}/training "
+        f"    ++exp_manager.max_time_per_run={params.timeout} "
+        f"    {extra_arguments} "
+    )
+    return cmd
+
+
+def train_rm(params: TrainingParams) -> str:
+    if params.config_name is None:
+        config_name = "dpo_config"
+    config_params = f""
+    training_script = "python -u /opt/NeMo-Aligner/examples/nlp/gpt/train_reward_model.py"
```

Comment:
I think it's better to keep the entry script on our side (by copy-pasting from nemo aligner). While your approach also works in the majority of cases, we don't want to put any restriction on what kind of container people can use our repo with. So while this is unlikely, someone can in principle have aligner installed in a different place or maybe they'd want to mount things inside container and then point python path there

Reply:
And same for the config, which also gives people option to provide their own (which otherwise isn't possible, since it needs to be the first argument, and so we cannot use our default way of passing parameters through the pipeline wrapper)

Reply:
Makes sense. Done :)